### ISSUE TITLE
Static variables inside a class throws

### ISSUE THREAD

**Issue by vincentvictoria** [javascript-obfuscator/javascript-obfuscator#510]
Static variables inside a class throws.

## Current Behavior

throws Unexpected token

## Minimal working example that will help to reproduce issue
    class TestStatic
        static A;
    }

Btw, thanks for fixing import.meta :)

**Comment by sanex3339** [comment]
Will see at another `acorn` plugin that will support this. 
Anyway, every `stage 3` feature should be supported by both: `acorn`, `escope`, `escodegen`. So, will check.

**Comment by sanex3339** [comment]
Sadly, currently, acorn 7 not supported static class fields, even with plugin.
So you have two ways:
- wait until this feature will move to stage 4
- transpile static properties with babel

**Comment by vincentvictoria** [comment]
Thanks for the thorough response.
I guess I will just fallback to `TestStatic.A = value` until the feature moves to stage 4.

**Comment by sanex3339** [comment]
I won't implement stage 3 features. Because for each feature 3 packages should support it:
`acorn`, `estraverse` and `escodegen`

**Comment by zNoobs** [comment]
What about now, is it supported?

**Comment by sanex3339** [comment]
No, until it's not supported by `acorn`, `estraverse` and `escodegen`

**Comment by adelin-mcbsoft** [comment]
The same issue seems to apply to properties (non-static variables), I suppose it has the same roots.
@zNoobs , have you found any workaround for it?

And the classy question for @sanex3339 : is there any projected date when these issues could be solved?

**Comment by sanex3339** [comment]
No any date

**Comment by delphir-com** [comment]
> The same issue seems to apply to properties (non-static variables), I suppose it has the same roots.
> @zNoobs , have you found any workaround for it?
> 
> And the classy question for @sanex3339 : is there any projected date when these issues could be solved?

Initially I also had issues with static variables and props.
I found a workaround which helped me to obfuscate reactjs codes, but I`m sure the same approach will work for a regular JS too.
In my case I had to "compile" jsx before providing it into javascript-obfuscator, but this compilation works for a regular JS classes too, so you can simply feed all your JS files into compiler, and then feed the result into javascript-obfuscator

```
npx babel --plugins @babel/proposal-class-properties --plugins @babel/plugin-transform-react-jsx /path/orig.js > /path/compiled.js
```

so at this point you get a `/path/compiled.js` file which can be easily accepted by javascript-obfuscator

**Comment by adelin-mcbsoft** [comment]
Hi @delphir-com ,

I confirm your approach works in JS as well, using _babel_'s "[plugin-proposal-class-properties](https://babeljs.io/docs/en/babel-plugin-proposal-class-properties)" plugin. 😃 
Hope it helps anyone who's struggling with this issue.
Thanks for your input and thanks @sanex3339 too for this awesome library,

Happy Coding!
Regards,
A.

**Comment by aaclayton** [comment]
+1 to this suggestion with some extra detail if it is helpful:

**Is your feature request related to a problem? Please describe.**
Public Class Fields are now supported by modern browsers and by Node.js 12+
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Public_class_fields

These field declarations are not currently supported by javascript obfuscator. Examples include:

```js
class ClassWithPublicFields {
  instanceField = "foo";
  static staticField = "bar";
}
```

Both `instanceField` and `staticField` will break the obfuscation process with an "Unexpected token " error.

**Describe the solution you'd like**
I would like for javascript-obfuscator to support these class fields and preserve their functionality in the obfuscated output.

**Describe alternatives you've considered**
Do not use these class fields in code which needs to be obfuscated.

**Additional context**
Please see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Public_class_fields for details and current browser/node compatibility.

I found this issue #510 which seems to be related, but was (perhaps incorrectly?) marked as closed. If this issue is blocked by an upstream dependency, it would be great to leave the issue open so it can continue to be tracked until such a time as the upstream blocker is removed. Thank you!

**Comment by sanex3339** [comment]
This issue will be closed in favor of 
https://github.com/javascript-obfuscator/javascript-obfuscator/issues/909

In short words: expect this feature soon.

**Comment by sanex3339** [comment]
Released as `2.18.0`
